### PR TITLE
gh-101123: Adapt vararg functions in the math module to Argument Clinic

### LINKED ISSUE #101123: invalid signature for math.hypot
```python
>>> import math
>>> import inspect
>>> inspect.signature(math.hypot) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sk/src/cpython/Lib/inspect.py", line 3295, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/inspect.py", line 3039, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/inspect.py", line 2531, in _signature_from_callable
    return _signature_from_builtin(sigcls, obj,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/inspect.py", line 2330, in _signature_from_builtin
    raise ValueError("no signature found for builtin {!r}".format(func))
ValueError: no signature found for builtin <built-in function hypot>
```

This patch works:
```diff
diff --git a/Modules/mathmodule.c b/Modules/mathmodule.c
index 1342162fa7..0e610eb9cb 100644
--- a/Modules/mathmodule.c
+++ b/Modules/mathmodule.c
@@ -2805,7 +2805,9 @@ math_hypot(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
 #undef NUM_STACK_ELEMS
 
 PyDoc_STRVAR(math_hypot_doc,
-             "hypot(*coordinates) -> value\n\n\
+             "hypot($module, *coordinates)\n\
+--\n\
+\n\
 Multidimensional Euclidean distance from the origin to a point.\n\
 \n\
 Roughly equivalent to:\n\
```

### Linked PRs
* gh-101124
* gh-126235

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-101123

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
I support using more AC in the stdlib.

File: Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix :func:`math.hypot` signature.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: IMO, we can drop the NEWS entry.

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
> I support using more AC in the stdlib.

+1, and so does the [faster CPython team](https://github.com/faster-cpython/ideas/issues/565).

File: Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst (hunk 1 above)
Comment (skirpichev) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: This is somewhat a user-visible change.

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: Yes, but [we normally do not produce NEWS entries for doc improvements](https://devguide.python.org/core-developers/committing/#changes-that-require-news-entries).

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: If you would like the NEWS entry, consider instead to put it in the _Documentation_ category.

Moreover, if you'd want, you can also create another NEWS entry for the Argument Clinic adaptions; we sometimes produce those. See the Git log and decide what you think makes most sense for your change :)

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: If you want to keep the NEWS entry as it is, give me a thumbs-up and I'll land this.

Comment (skirpichev) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: I don't think it's a documentation-only change.  The inspect module purpose is much wider.

But I'm not attached to this.  I'll just revert the news entry.

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2024-10-31-14-45-52.gh-issue-101123.XRpEJs.rst]: Fair enough.

**Comment by erlend-aasland** [comment]
Thanks, Sergey!

**Pull request merged**